pkg/kp/rollstore/consul_store.go

```diff
@@ -521,35 +521,75 @@ func (s consulStore) Lock(id roll_fields.ID, session string) (bool, error) {
 }
 
 func (s consulStore) Watch(quit <-chan struct{}) (<-chan []roll_fields.Update, <-chan error) {
-	outCh := make(chan []roll_fields.Update)
-	errCh := make(chan error)
 	inCh := make(chan api.KVPairs)
 
+	outCh, errCh := publishLatestRolls(inCh, quit)
 	go consulutil.WatchPrefix(rollTree+"/", s.kv, inCh, quit, errCh)
 
+	return outCh, errCh
+}
+
+func publishLatestRolls(inCh <-chan api.KVPairs, quit <-chan struct{}) (<-chan []roll_fields.Update, chan error) {
```

Comment:
yeah because it also gets passed to the roll watch (two publishers). Weird, I know, but not something i wanted to tackle in this PR